src/Common/src/System/PasteArguments.cs

```diff
@@ -80,6 +80,17 @@ private static void AppendArgument(StringBuilder stringBuilder, string argument)
             }
         }
 
+        internal static void AppendArgument(StringBuilder stringBuilder, Collection<string> argumentList)
```

Comment:
The name should be AppendArguments (plural) and I think it's more appropriate for this helper to live inside the Process code itself. I don't see it being as general purpose as the StringBuilder+string overload.